app/utils/keycloak.ts

```diff
@@ -0,0 +1,16 @@
+import getConfig from 'next/config';
+import Keycloak from 'keycloak-js';
+
+const { publicRuntimeConfig = {} } = getConfig() ?? {};
+const { sso_client_id, sso_url } = publicRuntimeConfig;
+
+const keycloak = new Keycloak({
+  url: sso_url,
+  realm: 'standard',
+  clientId: sso_client_id,
+});
+
+keycloak.onAuthRefreshError = () => keycloak.logout();
+
+keycloak.onTokenExpired = () => console.log('expired?');
```

Comment:
Maybe a more clear log message would be useful hear?

Reply:
oh good catch, ill remove that was for testing